src/System.Private.Xml/src/System/Xml/Serialization/PrimitiveXmlSerializers.cs

```diff
@@ -2,10 +2,21 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+#if XMLSERIALIZERGENERATOR
+namespace Microsoft.XmlSerializer.Generator
+#else
 namespace System.Xml.Serialization
+#endif
 {
+    using System;
+
+#if XMLSERIALIZERGENERATOR
+    internal class XmlSerializationPrimitiveWriter : Microsoft.XmlSerializer.Generator.XmlSerializationWriter
+#else
     internal class XmlSerializationPrimitiveWriter : System.Xml.Serialization.XmlSerializationWriter
+#endif
     {
+
```

Comment:
We probably can combine the two #if sections into one and put the using outside.